main.go

```diff
@@ -22,51 +22,73 @@ func main() {
 
 	// Iterate over each pull request
 	for _, pr := range prs {
-		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
-
-		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
-		diff, err := client.GetPrDiff(pr)
-		if err != nil {
-			log.Fatalf("Failed to get raw diff: %v", err)
+		comments := reviewPR(pr, client)
+		commitId := pr.GetHead().GetSHA()
+		for _, comment := range comments {
+			// post to github
+			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
+			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
+			err := client.PostComment(pr, ghComment)
+			if err != nil {
+				log.Fatalf("Failed to post comment: %v", err)
+			}
 		}
+	}
+}
+
+func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
+	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
+
+	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
+	diff, err := client.GetPrDiff(pr)
+	if err != nil {
+		log.Fatalf("Failed to get raw diff: %v", err)
+	}
+
+	diffTransformer := transformer.DiffTransformer{}
+	diffTransformer.Transform(diff)
 
-		diffTransformer := transformer.DiffTransformer{}
```

Comment:
Missing Test: Unit tests could be added to ensure correct posting of comments to a pull request